main.c

```diff
@@ -230,3 +377,76 @@ static void __interrupt() interrupt_handler(void) {
         PIR3bits.TMR0IF = 0;
     }
 }
+
+//Motor control functions
+void pwm_init(void){
```

Comment:
I would put this in a different source file for cleanliness, and also cut out the test lines. The steps from the datasheet are good and you should keep them